? XmlSearch/unxlngx6.pro
cvs diff: Diffing .
cvs diff: Diffing XmlSearch/prj
cvs diff: Diffing XmlSearch/src
cvs diff: Diffing XmlSearch/src/com
cvs diff: Diffing XmlSearch/src/com/sun
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch
Index: XmlSearch/src/com/sun/xmlsearch/makefile.mk
===================================================================
RCS file: /cvs/util/XmlSearch/src/com/sun/xmlsearch/makefile.mk,v
retrieving revision 1.5
diff -u -r1.5 makefile.mk
--- XmlSearch/src/com/sun/xmlsearch/makefile.mk	25 Oct 2005 11:11:58 -0000	1.5
+++ XmlSearch/src/com/sun/xmlsearch/makefile.mk	2 Nov 2005 14:32:12 -0000
@@ -21,7 +21,7 @@
 
 .INCLUDE : settings.mk
 
-JARFILES 		=  jaxp.jar parser.jar xt.jar
+JARFILES 		=  jaxp.jar parser.jar xt.jar xercesImpl.jar
 #EXTRAJARFILES 		=  jaxp.jar parser.jar xt.jar  
 
 JAVAFILES		= $(subst,$(CLASSDIR)$/$(PACKAGE)$/, $(subst,.class,.java $(JAVACLASSFILES))) 
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/db
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/gui
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/indexer
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/qe
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/servlet
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/tree
Index: XmlSearch/src/com/sun/xmlsearch/tree/TreeBuilder.java
===================================================================
RCS file: /cvs/util/XmlSearch/src/com/sun/xmlsearch/tree/TreeBuilder.java,v
retrieving revision 1.2
diff -u -r1.2 TreeBuilder.java
--- XmlSearch/src/com/sun/xmlsearch/tree/TreeBuilder.java	9 Sep 2005 16:41:49 -0000	1.2
+++ XmlSearch/src/com/sun/xmlsearch/tree/TreeBuilder.java	2 Nov 2005 14:32:14 -0000
@@ -46,7 +46,9 @@
 import org.xml.sax.HandlerBase;
 import org.xml.sax.AttributeList;
 import org.xml.sax.InputSource;
-import com.sun.xml.parser.Parser;
+import org.xml.sax.SAXNotRecognizedException;
+import org.xml.sax.SAXNotSupportedException;
+import org.apache.xerces.parsers.SAXParser;
 import com.sun.xmlsearch.util.IntegerArray;
 import com.jclark.xsl.om.*;
 
@@ -144,6 +146,7 @@
 
 		public URL getURL() { return null; }
 		public int getLineNumber() { return -1; }
+        public int getColumnNumber() { return -1; }
 		public NamespacePrefixMap getNamespacePrefixMap() {
 			return _parent.getNamespacePrefixMap();
 		}
@@ -153,6 +156,12 @@
 		public String getUnparsedEntityURI(String name) {
 			return "getUnparsedEntityURI";
 		}
+        public String getPublicId() {
+            return "getPublicId not implemented";
+        }
+        public String getSystemId() {
+            return "getSystemId not implemented";
+        }
   
 		public boolean isLeaf() { return true; }
 		public int getChildCount() { return 0; }
@@ -612,7 +621,7 @@
 
     private static final int InitStackSize = 256;
     private static final Name EmptyName = null;
-    private Parser _parser = new Parser();
+    private SAXParser _parser = new SAXParser();
     private Hashtable _names = new Hashtable(256);
   
     private NameTable _nameTable;
@@ -625,7 +634,12 @@
     public TreeBuilder(NameTable nameTable) {
 		_nameTable = nameTable;
 		_nsMap = _nameTable.getEmptyNamespacePrefixMap();
-		_parser.setFastStandalone(true);
+        try {
+            _parser.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+            _parser.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        } catch (SAXNotRecognizedException e) {
+        } catch (SAXNotSupportedException e) {
+        }
     }
 
     public Node makeSubstituteElement(Name name, Node2[] children, Node original) {
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/util
Index: XmlSearch/src/com/sun/xmlsearch/util/Configuration.java
===================================================================
RCS file: /cvs/util/XmlSearch/src/com/sun/xmlsearch/util/Configuration.java,v
retrieving revision 1.2
diff -u -r1.2 Configuration.java
--- XmlSearch/src/com/sun/xmlsearch/util/Configuration.java	9 Sep 2005 16:45:09 -0000	1.2
+++ XmlSearch/src/com/sun/xmlsearch/util/Configuration.java	2 Nov 2005 14:32:14 -0000
@@ -40,8 +40,7 @@
 import java.util.Vector;
 import java.net.URL;
 import org.w3c.dom.*;
-import com.sun.xml.tree.XmlDocument;
-import com.sun.xml.parser.Resolver;
+import org.apache.xerces.parsers.DOMParser;
 import org.xml.sax.InputSource;
 
 public final class Configuration {
@@ -68,15 +67,18 @@
     }
 
     public static Element parse(URL location) throws Exception {
-	InputSource input = Resolver.createInputSource(location, true);
-	XmlDocument document = XmlDocument.createXmlDocument(input, false);
-	return document.getDocumentElement();
+        DOMParser parser = new DOMParser();
+        parser.setFeature("http://xml.org/sax/features/validation", true);
+        parser.parse(location.toString());
+        return parser.getDocument().getDocumentElement();
     }
 
     public static Element parse(byte[] xmlBytes) throws Exception {
-	ByteArrayInputStream input = new ByteArrayInputStream(xmlBytes);
-	XmlDocument document = XmlDocument.createXmlDocument(input, false);
-	return document.getDocumentElement();
+        ByteArrayInputStream input = new ByteArrayInputStream(xmlBytes);
+        DOMParser parser = new DOMParser();
+        parser.setFeature("http://xml.org/sax/features/validation", false);
+        parser.parse(new InputSource(input));
+        return parser.getDocument().getDocumentElement();
     }
 
     /** arg: Element that can be direct parent of PrefixTranslation-s
@@ -97,4 +99,5 @@
 	else
 	return null;
     }
+
 }
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/xml
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/xml/indexer
Index: XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexBuilder.java
===================================================================
RCS file: /cvs/util/XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexBuilder.java,v
retrieving revision 1.7
diff -u -r1.7 XmlIndexBuilder.java
--- XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexBuilder.java	9 Sep 2005 16:52:56 -0000	1.7
+++ XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexBuilder.java	2 Nov 2005 14:32:15 -0000
@@ -42,15 +42,15 @@
 import java.net.URL;
 import org.xml.sax.InputSource;
 import org.xml.sax.HandlerBase;
-import com.sun.xml.parser.Resolver;
-import com.sun.xml.tree.XmlDocument;
+import org.xml.sax.SAXNotRecognizedException;
+import org.xml.sax.SAXNotSupportedException;
 import com.sun.xmlsearch.tree.*;
 import com.sun.xmlsearch.util.*;
 import com.sun.xmlsearch.db.*;
 import com.sun.xmlsearch.xml.XmlIndex;
 
-import com.sun.xml.parser.Parser;
-import com.sun.xml.parser.ValidatingParser;
+import org.apache.xerces.parsers.DOMParser;
+import org.apache.xerces.parsers.SAXParser;
 
 import com.jclark.xsl.om.*;
 import com.jclark.xsl.sax.*;
@@ -433,7 +433,7 @@
 	private boolean _useLocalTransformFile = false;
 
 	private TreeBuilder _treeBuilder;
-	private MyXslEngine _transformEngine;
+    private XSLTransformEngine _transformEngine;
 	private Transform _indexingTransform;
 	private Transform _defaultTransform;
 	private IndexAdapter _indexAdapter;
@@ -465,16 +465,21 @@
 	public void indexDocument(URL docURL, String title) throws Exception {
 		InputSource source = new InputSource(docURL.openStream());
 		source.setSystemId(docURL.toString());
-		Parser sourceParser = new Parser();
-		sourceParser.setFastStandalone(true);
+        SAXParser sourceParser = new SAXParser();
+        try {
+            sourceParser.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+            sourceParser.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        } catch (SAXNotRecognizedException e) {
+        } catch (SAXNotSupportedException e) {
+        }
 		XMLProcessorEx sourceLoader = new XMLProcessorImpl(sourceParser);
 		//	long start = System.currentTimeMillis();
 		Node root =
 			sourceLoader.load(
 				source,
 				0,
-				_transformEngine.getSourceLoadContext(),
-				_transformEngine.getNameTable());
+                null,
+                null);
 
 		//System.out.println((System.currentTimeMillis()-start)+" msec parse");
 		// build association from generated node ids to nodes
@@ -518,7 +523,7 @@
 
 	private final void initXmlProcessor(String transform) throws Exception {
 		_transformEngine = new MyXslEngine();
-		NameTable nameTable = _transformEngine.getNameTable();
+        NameTable nameTable = new NameTableImpl();
 		_treeBuilder = new TreeBuilder(nameTable);
 		_defaultTransform = getTransform("default");
 		_indexingTransform = getTransform(transform);
@@ -545,9 +550,9 @@
 		}
 		// System.out.println(stylesheetUrl.toString());
 		InputStream stylesheetStream = stylesheetUrl.openStream();
-		XmlDocument sheet =
-			XmlDocument.createXmlDocument(stylesheetStream, false);
-		return _transformEngine.createTransform(sheet);
+        DOMParser parser = new DOMParser();
+        parser.parse(new InputSource(stylesheetStream));
+        return _transformEngine.createTransform(parser.getDocument());
 	}
 
 	// GTM new:
Index: XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexer.java
===================================================================
RCS file: /cvs/util/XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexer.java,v
retrieving revision 1.2
diff -u -r1.2 XmlIndexer.java
--- XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexer.java	9 Sep 2005 16:53:09 -0000	1.2
+++ XmlSearch/src/com/sun/xmlsearch/xml/indexer/XmlIndexer.java	2 Nov 2005 14:32:16 -0000
@@ -39,8 +39,6 @@
 import java.io.*;
 import java.util.*;
 
-import com.sun.xml.tree.XmlDocument;
-import com.sun.xml.parser.Resolver;
 import org.w3c.dom.*;
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
cvs diff: Diffing XmlSearch/src/com/sun/xmlsearch/xml/qe
